src/ast/ddl.rs

```diff
@@ -247,6 +249,11 @@ pub enum TableConstraint {
         name: Option<Ident>,
         expr: Box<Expr>,
     },
+    Key {
```

Comment:
Can you please add comments here about what this field means / how it maps back to SQL?

Reply:
I'm almost sure this is related to MySQL's index creation [1](https://dev.mysql.com/doc/refman/8.0/en/create-table.html). That's something I wanted to have as well.

Conceptually has the same idea of PostgreSQL's CREATE INDEX [2](https://www.postgresql.org/docs/current/sql-createindex.html) statement and MySQL's [3](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) one, but inside the create table.

In MySQL seems better to have it inside the CREATE TABLE, since their CREATE INDEX doesn't support `IF NOT EXISTS`, while the CREATE TABLE does.

[1]: https://dev.mysql.com/doc/refman/8.0/en/create-table.html
[2]: https://www.postgresql.org/docs/current/sql-createindex.html
[3]: https://dev.mysql.com/doc/refman/8.0/en/create-index.html

Reply:
@RainJoe please LMK if I'm wrong, but `UNIQUE KEY` isn't the same as a `UNIQUE` constraint, while only `KEY` refers to the same as `CREATE INDEX`?

Wouldn't be better to keep `UNIQUE [KEY | INDEX]` as a `UNIQUE` table constraint?

Reply:
Yes, it's not same. I have fixed it, PTAL.